### PR TITLE
CRUD of Locations, WorkSpaces

### LINKED ISSUE #52: CRUD Location & Workspaces

### PR DESCRIPTION
**Pull request by vineetvk01**
APIs to perform CRUD operations on location and workspaces 
Closes #52

### CODE REVIEW

**Review by nitishmittal1990 — COMMENTED** [review]
Please Tag Issue.

**Comment by vineetvk01** [comment]
@fenilgandhi  Please provide MONGODB_URL in .env file for complete CI testing.

File: server/src/controllers/workspace.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+    // Place current logged in user's ID --After User Auth merge
+    const user = { _id: "5e4700a5ffcc4113d83a80ff" };
+    logger.info("Owner's User Id : 5e4700a5ffcc4113d83a80ff");
+
+    const addressSchema = {
+      ...address
+    };
+
+    // logger.debug(`Address received: ${address}`);
```
Comment (nitishmittal1990) [inline server/src/controllers/workspace.js]: Remove comments

**Review by nitishmittal1990 — APPROVED** [review]
LGTM!

File: server/src/controllers/workspace.js (hunk 1 above)
Comment (vineetvk01) [reply server/src/controllers/workspace.js]: Removed the comments

File: server/src/__tests__/controller/workspace.test.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+/* eslint-disable */
```
Comment (rishabh1403) [inline server/src/__tests__/controller/workspace.test.js:1]: eslint disable ?

File: server/src/controllers/workspace.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
+import Logger from '../logger';
+import Location from '../models/location';
+import Workspace from '../models/workspace';
+
+const logger = Logger('controller > workspace');
+
+exports.getWorkspaces = async (request, response, next) => {
+  const { qLocation, page = 1, count = 10 } = request.params;
```
Comment (rishabh1403) [inline server/src/controllers/workspace.js:8]: too much logic in one function. 
Break thr function into simpler components

File: server/src/controllers/workspace.js [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+      workspaces: arrayOfWorkspaces.slice(startIndex, endIndex),
+    });
+  } catch (error) {
+    logger.error(error);
+    response.status(500);
+  }
+  return next();
+};
+
+exports.addWorkspaceImages = async (request, response, next) => {
```
Comment (rishabh1403) [inline server/src/controllers/workspace.js:34]: function too complex

**Review by rishabh1403 — APPROVED** [review]
Couple of functions are too complex. Please break into smaller functions

File: server/src/__tests__/controller/workspace.test.js (hunk 2 above)
Comment (vineetvk01) [reply server/src/__tests__/controller/workspace.test.js:1]: Test files shows eslint issues. Like:  describe/test function are not declared.

File: server/src/controllers/workspace.js (hunk 3 above)
Comment (vineetvk01) [reply server/src/controllers/workspace.js:8]: Done!

File: server/src/controllers/workspace.js (hunk 4 above)
Comment (vineetvk01) [reply server/src/controllers/workspace.js:34]: Distributed chunks of operation in schema statics functions

**Pull request merged**